doc/rules/php_unit/php_unit_data_provider_return_type.rst

```diff
@@ -2,7 +2,13 @@
 Rule ``php_unit_data_provider_return_type``
 ===========================================
 
-The type returned by data provider must be ``iterable``.
+The return type of PHPUnit data provider must be ``iterable``.
+
+Description
+-----------
+
+Data provider must return ``iterable``, either an array of arrays or an object
```

Comment:
please add a rationale why it should be this way / why fixer should be used

Reply:
I can't think of anything more than is in the summary. The is no general reason to use it other than to have unified code base - which I guess applies to 90% of the fixers.

TBH I disagree with the enforcement of the description added in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7128 (BTW I do not see any approval there 😉) - I cannot think for most of the fixers added there as exceptions about anything to have in the description other than what is the summary.

Reply:
in `summary`, you say "it should return `iterable`". But JohnDoe, imagine a junior, will not have any clue why. In `description`, I want to have reasoning why/when this fixer should be used.

> BTW I do not see any approval there 😉

I believe I got some superpowers after 10 years in this repo 🤷🏻

Reply:
> I want to have reasoning why/when this fixer should be used.

I don't have any :(

Reply:
To be honest I also don't like merging without approvals. I understand @keradus is a project lead, but I believe there's always place for discussion about the vision 🙂.

Reply:
Especially with changes that are affecting the repo hard. Besides - as I already wrote - I disagree with the change, it increases the difficulty of adding a new fixer.

Like in [feature: AlphabeticalArrayKeySortFixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/5492), the [description](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/5492/files#diff-5dea671198fb463fcf94408fb01ac28d10b77ca9ae49ab0bb7e469b60619bb6fR57) is not what you, @keradus wrote: "I want to have reasoning why/when this fixer should be used."

Reply:
to trivialize the example - wouldn't it make sense to not only say "use === instead of ==", but also guide ppl *why* to do so?

Reply:
How would you explain people why they should have ordered types? Or why they should have single empty line after declaring strict types? There are so many fixers that do things "just because" some group of people decided it should be a standard - what's the point in providing reasoning, if anyone can make their own reasoning out of the way how fixer works? Often it's just a matter of taste, not any particular reason.

Reply:
often that taste is based on alignment - ie there were times when we literally read each line of PSR and made dedicated fixer for them.

eg https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7157

Reply:
Yeah, but fixer being part of PSR/PER ruleset is enough reasoning in my opinion, there's no need for clarify it in the description too.

Reply:
and JohnDoe, having a look at `method_argument_space` only (eg because it was telling him that his codebase does not follow this rule, that is configured in project) has no clue about that.
Eg, `php-cs-fixer descbire` is not mentioning the ruleset, docs is claiming 4 different rulesets, and to understand the ultimate reason of this rule in first place (eg which community align on it first) - they need to go to codebase and start looking ,hoping for that.

ultimately, I could agree that it's non-obvious for some rules to say "why" other than "given community aligned on it". But we often don't say even that.
And for things like "=== vs ==", we do not tell any reason, only enforcing it, without supporting JohnDoe to understand it.

Imagine you ask junior dev in your team to follow a practice. whatever practice. if you enforce it on them from position of power, they will follow, but not be happy about it, and not liking it. But if you give them the reasoning (other than "do as I command"), they will understand it and see the value.

Reply:
I understand it, I just don't think enforcing such description is good, because it can lead to redundant and/or not helpful information 😉.

Reply:
please provide any other way.

Reply:
For existing fixers:
1. Review existing fixers' descriptions
2. Create an issue with a todo list of fixers to improve
3. Provide PR(s) with improvements
4. Close the issue when every and each fixer is described properly

For new fixers: take care of it on code review stage.